### ISSUE TITLE
Create Documentation for `TokenResolver` feature

### ISSUE THREAD

**Issue by tilucasoli** [conceptadev/mix#536]
### Description:
Develop comprehensive documentation for `TokenResolver` feature. This documentation should explain the purpose, usage, and key methods or properties of these classes, providing clarity for developers on how they function and can be extended or utilized effectively.

#### User Story:
**_As a_** developer using the framework,
**_I want to_** understand the functionality and usage of classes implementing `WithTokenResolver`,
**_so that_** I can confidently work with or extend these classes in my projects.

### Acceptance Criteria:

- [ ] Identify and list all classes that implement `WithTokenResolver`.
- [ ] Document the purpose of this feature and how it impacts the theme creation
- [ ] The documentation should include:
  - Key methods and their functionality.
  - Examples of how to use the class.
- [ ]  Include inline comments in the codebase where necessary for developers.
- [ ]  Publish the documentation to the official developer documentation site or repository.

**Comment by lvsecoto** [comment]
Yes! We also need to Resolve token as SVG-icon, string resource...



This is the greatest Style System . I have used it in my company new UI-Framework(Not a Material-like theme)!

**Comment by tilucasoli** [comment]
> Yes! We also need to Resolve token as SVG-icon, string resource...
> 
> This is the greatest Style System . I have used it in my company new UI-Framework(Not a Material-like theme)!

Could you please create an issue with the example needed? We require a bit more detail to assist you.

**Comment by lvsecoto** [comment]
> > Yes! We also need to Resolve token as SVG-icon, string resource...
> > This is the greatest Style System . I have used it in my company new UI-Framework(Not a Material-like theme)!
> 
> Could you please create an issue with the example needed? We require a bit more detail to assist you.

Hi~ 

For now there fixed type of token can be passed to MixThemeData : 
```
  const MixThemeData.raw({
    required this.textStyles,
    required this.colors,
    required this.breakpoints,
    required this.radii,
    required this.spaces,
    this.defaultOrderOfModifiers,
  });
```

But may be we need these type's examples like:

```
StyledTokens<MixToken<Svg>, Svg> svgs,
StyledTokens<MixToken<String>, String> stringResources,
```

I would like to design my own token type. Thank!

**Comment by tilucasoli** [comment]
Just wanted to follow up on this one — really love the direction this thread took, and the use case @lvsecoto raised (custom token types like `Svg` and string resources, beyond the fixed set in `MixThemeData.raw`) is a great one. 🙌

Good news: this is effectively resolved in **Mix v2**. The token system was reworked around `MixToken` + `MixScope`, and there's now proper documentation for it:

- **Design Tokens guide** — covers declaring tokens, providing values via `MixScope`, the built-in token types, and resolving tokens programmatically with `$token.resolve(context)`: https://www.fluttermix.com/documentation/mix/guides/design-token
- **Creating Custom Tokens tutorial** — exactly the "design my own token type" case from this thread, for values not covered by the built-ins: https://www.fluttermix.com/documentation/mix/tutorials/creating-custom-tokens

Heads up that the v1 `TokenResolver` / `WithTokenResolver` classes aren't part of the v2 API surface anymore — the resolution flow moved into `MixScope`/`MixToken` — so the v2 docs above are the place to look rather than docs for those specific classes.

Thanks for the thoughtful discussion here, and for the kind words about the style system. Closing the loop since v2 covers this. 🎉
